src/ts/deploy.ts

```diff
@@ -292,9 +292,18 @@ export async function prepareVirtualDeploymentFromSafe(
 
 export async function getDeployArgsFromRealToken(
   realToken: Contract,
-): Promise<Omit<RealTokenDeployParams, "totalSupply">> {
+): Promise<RealTokenDeployParams> {
+  const filterMintingTransfers = realToken.filters.Transfer(
+    constants.AddressZero,
+    null,
+    null,
+  );
+  const logs = await realToken.queryFilter(filterMintingTransfers, 0, "latest");
+  const events = logs.map((log) => realToken.interface.parseLog(log));
+  const totalSupply = BigNumber.from(events[0].args.value).toString();
```

Comment:
This is the trick, right? We use the emitted event to get the total supply instead of passing it as a param (clever!). This isn't really clear from the PR description (the linked PR is mainly unrelated to this trick).

Reply:
Correct! I fixed the PR description to better explain what is going on and to make it look less "insidery".